src/components/Charts/ChataChart/ChataChart.js

```diff
@@ -192,9 +189,32 @@ export default class ChataChart extends Component {
     return { chartHeight, chartWidth }
   }
 
+  sortChartDataByDate = (props) => {
+    try {
+      if (!props.data || typeof props.data !== 'object') {
+        return undefined
+      }
+      const dateColumnIndex = getDateColumnIndex(props.columns)
+      if (dateColumnIndex >= 0) {
+        let sortedData = [...props.data].sort((a, b) =>
+          props.dateSortFn(a[dateColumnIndex], b[dateColumnIndex], true)
+        )
+        return sortedData
+      }
+
+      return props.data
+    } catch (error) {
+      console.error(error)
+      return props.data
+    }
+  }
+
   aggregateRowData = (props) => {
     const { stringColumnIndex, numberColumnIndices } = props
-    const sortedData = _sortBy(props.data, (row) => row?.[stringColumnIndex])
+    const sortedData = this.sortChartDataByDate(props)
```

Comment:
This probably won't work the way you are expecting it to. The reason we have to sort this by `stringColumnIndex` is because when the aggregation loops through the data it compares the value to the previous row and if its the same it adds them together. The stringColumnIndex is the column that is on the chart axis, which is the column we want to aggregate by.

In other words, the chart will not always have the `DATE` column on the axis. Especially because we allow the user to choose a different column in the axis selector dropdown. Once the user selects a different ordinal axis, the data will have to be aggregated again by that new column.



Instead of changing the column to sort by, can you detect what type of column the `stringColumnIndex` is, and sort it according to that?